pkg/ddevapp/laravel.go

```diff
@@ -18,35 +19,31 @@ func isLaravelApp(app *DdevApp) bool {
 	return fileutil.FileExists(filepath.Join(app.AppRoot, "artisan"))
 }
 
-func envSettingsWarning(status int) {
-	var srcFile = ".env"
-	var message = "Don't forget to configure the database in your .env file"
-
-	if WarnTypeAbsent == status {
-		srcFile += ".example"
-		message = "Don't forget to create the .env file with proper database settings"
-	}
-	util.Warning(message)
-	util.Warning("You can do it with this one-liner:")
-	util.Warning("ddev exec \"cat %v | sed  -E 's/DB_(HOST|DATABASE|USERNAME|PASSWORD)=(.*)/DB_\\1=db/g' > .env\"", srcFile)
-	util.Warning("Read more on https://ddev.readthedocs.io/en/stable/users/quickstart/#laravel")
-}
-
 func laravelPostStartAction(app *DdevApp) error {
-	if fileutil.FileExists(filepath.Join(app.AppRoot, ".env")) {
-		isConfiguredDbHost, err := fileutil.FgrepStringInFile(app.SiteSettingsPath, `DB_HOST=db`)
-		isConfiguredDbConnection, _ := fileutil.FgrepStringInFile(app.SiteSettingsPath, `DB_CONNECTION=ddev`)
-		if err == nil && !isConfiguredDbHost && !isConfiguredDbConnection {
-			envSettingsWarning(WarnTypeNotConfigured)
+	envContents, err := ReadEnvFile(app)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("Unable to read .env file: %v", err)
+	}
+	if os.IsNotExist(err) {
+		err = fileutil.CopyFile(filepath.Join(app.AppRoot, ".env.example"), filepath.Join(app.AppRoot, ".env"))
+		if err != nil {
+			return err
+		}
+		envContents, err = ReadEnvFile(app)
+		if err != nil {
+			return err
 		}
-	} else {
-		envSettingsWarning(WarnTypeAbsent)
+	}
+	envContents["DB_HOST"] = "db"
+	envContents["DB_PORT"] = "3306"
+	envContents["DB_DATABASE"] = "db"
+	envContents["DB_USERNAME"] = "db"
+	envContents["DB_PASSWORD"] = "db"
+	envContents["DB_CONNECTION"] = "ddev"
```

Comment:
Setting the "DB_CONNECTION" here is invalid for Laravel unless clients add a DDEV section to `./config/database.php`. That requires a big chunk of code. 

The solution I came up with for the action PR (https://github.com/drud/ddev/pull/3954/files) is more elegant I think; use the database name. This is more future proof too, if Laravel updates the database options (strict, charset etc.).

Note that "postgres" uses a "pgsql" in the laravel config file.

Reply:
So great that you have studied this so well! I think we'll have to talk in Discord for me to understand the big picture.